integrationtests/self/http_test.go

```diff
@@ -957,3 +957,118 @@ func TestHTTPStreamer(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, PRData, repl)
 }
+
+func TestHTTPClientTrace(t *testing.T) {
```

Comment:
Would it be useful to test the order of the events? You could pass them into an event channel.

Reply:
good idea!

Reply:
fixed